feat: add Venly connector to packages

davidzwfu opened the pull request:
### Description
Hello, David from the Venly team here. 
Opening this PR to get our Venly wallet connector added to the list of packages.


### Checklist
- [x] Increment the version field in `package.json` of the package you have made changes in following [semantic versioning](https://semver.org/) and using alpha release tagging
- [x] Check the box that allows repo maintainers to update this PR
- [x] Test locally to make sure this feature/fix works
- [x] Run `yarn check-all` to confirm there are not any associated errors
- [x] Confirm this PR passes Circle CI checks
- [x] Add or update relevant information in the documentation

### If this PR includes changes to add an injected wallet or SDK wallet module: 
Please complete the following using the internal demo package.
To run this demo use the command `yarn && yarn dev` to get the project running at `http://localhost:8080/`

#### Tests with demo app (SDK)
- [x] send transaction
- [x] switch chains
- [x] sign message
- [x] sign typed message
- [x] disconnect

leightkt commented:
@davidzwfu Will you please also include a README for Venly for the docs site: [https://github.com/blocknative/web3-onboard/tree/develop/docs/src/routes/docs/%5B...4%5Dwallets](https://github.com/blocknative/web3-onboard/tree/develop/docs/src/routes/docs/%5B...4%5Dwallets)

davidzwfu commented:
> @davidzwfu Will you please also include a README for Venly for the docs site: https://github.com/blocknative/web3-onboard/tree/develop/docs/src/routes/docs/%5B...4%5Dwallets

Added

Adamj1232 commented on docs/src/routes/docs/[...4]wallets/venly.md:
```diff
@@ -0,0 +1,55 @@
+# Coinbase
```
```suggestion
# Venly
```

Adamj1232 commented on packages/venly/package.json:
```diff
@@ -0,0 +1,51 @@
+{
+  "name": "@web3-onboard/venly",
+  "version": "2.0.0",
```
```suggestion
  "version": "2.0.0-alpha.1",
```

Adamj1232 commented on packages/venly/src/index.ts:
```diff
@@ -0,0 +1,30 @@
+import { WalletInit } from '@web3-onboard/common';
+import { VenlyProvider, VenlyProviderOptions, SECRET_TYPES } from '@venly/web3-provider';
```
Can we import this dynamically within the getInterface method below?

davidzwfu commented on packages/venly/src/index.ts:
```diff
@@ -0,0 +1,36 @@
+import { WalletInit } from '@web3-onboard/common';
+
+interface VenlyProviderOptions {
```
Updated readme and docs with the additional info

Adamj1232 commented on packages/venly/src/index.ts:
```diff
@@ -0,0 +1,36 @@
+import { WalletInit } from '@web3-onboard/common';
+
+interface VenlyProviderOptions {
+  clientId: string
+  secretType?: any
+  environment?: string
+}
```
```suggestion
interface VenlyProviderOptions {
  /**
  * Multiline example:
  * ClientId is a required prop and can be created at vely.com
  */
  clientId: string
  /* Single line example: secretType is optional and does this....
  secretType?: any
  environment?: string
}
```
Can you please add typescript notes to each property on what it does?

davidzwfu replied:
Added typescript note for clientId.
I opted to remove secretType and environment to keep things simpler, since those can be derived from the chain property anyway.

Adamj1232 commented on packages/venly/README.md:
````diff
@@ -0,0 +1,44 @@
+## Options
+
+```typescript
+type VenlyProviderOptions = {
````
Look like secretType is missing

davidzwfu replied:
Updated readme as well

Adamj1232 commented on docs/src/routes/docs/[...4]wallets/venly.md:
````diff
@@ -0,0 +1,60 @@
+
+```typescript
+type VenlyProviderOptions = {
+  clientId: string
````
Looks like secretType is missing

davidzwfu replied:
Updated to match

Adamj1232 commented on packages/venly/package.json:
```diff
@@ -0,0 +1,51 @@
+    "typescript": "^4.5.5"
+  },
+  "dependencies": {
+    "@venly/web3-provider": "^3.0.0-develop.8",
```
Should this be pointed to a develop version? 
If there are any env build/bundler configs for  the @venly package that would be helpful to document in our core docs here: https://github.com/blocknative/web3-onboard/blob/develop/docs/src/routes/docs/%5B...3%5Dmodules/core.md#build-environments
For official release out of an `alpha` package the package will need to be able to be run within all of our example projects: https://github.com/blocknative/web3-onboard/blob/develop/docs/src/routes/docs/%5B...3%5Dmodules/core.md#build-environments
Testing for this will need to be done after `alpha` package publishing to get real-world instantiation/integration testing.

davidzwfu replied:
Updated to latest stable build

Adamj1232 commented:
@davidzwfu Can you give me write access for the web3-onboard fork so I can make some formatting updates to the fork?

Adamj1232 commented:
@davidzwfu the Venly package doesnt work (and it seems wasnt tested) with the internal demo app `packages/demo` that will start using `yarn dev`
The error thrown is ` | ERROR in ../../node_modules/micro-ftch/index.js 56:17-32
 | Module not found: Error: Can't resolve 'zlib' in '/Users/adam/blocknative/community-forks/venly/web3-onboard/node_modules/micro-ftch'
 | 
 | BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
 | This is no longer the case. Verify if you need this module and configure a polyfill for it.`

Adamj1232 commented:
@davidzwfu  friendly ping on the error noted above...any progress here? At the end of the week this PR will be flagged as stale if no comments or updates are made from your end.

davidzwfu commented:
> Reference

@Adamj1232 Apologies for the delay, the team and I have been getting our newest builds ready for release. I've already tested the Venly connector on the packages/demo app, and will be submitting all the changes along with the latest @venly/web3-provider build once that's ready.

Adamj1232 commented:
@davidzwfu thanks for updating! Can you set me up with a blocknative clientId for internal demos?

davidzwfu commented:
> @davidzwfu thanks for updating! Can you set me up with a blocknative clientId for internal demos? Also, can you please merge in the latest `develop` branch from bn when you have a chance.

Merged. I've sent you an email regarding the clientId

Adamj1232 commented:
> > @davidzwfu thanks for updating! Can you set me up with a blocknative clientId for internal demos? Also, can you please merge in the latest `develop` branch from bn when you have a chance.
> 
> Merged. I've sent you an email regarding the clientId

Thanks @davidzwfu  - No how do I test with Venly after setting up an account? Do I need to change the Venly env to production?
Can we scope the clientId to specific origins for our demo apps pointing to production?

Adamj1232 commented:
I get an error like this with my pin of 6 digits 
<img width="375" alt="Screenshot 2023-05-16 at 10 09 33 AM" src="https://github.com/blocknative/web3-onboard/assets/24457880/f27a449c-bf48-4134-ba0d-fcec3011c9f7">
Also, when my pin was set to 4 digits I got a similar error that said it needed to be 6 digits
Couldn't connect to the account I made through the main Venly page at all that had a wallet setup (maybe production)
Doesnt seem to be an issue with the web3-onboard wallet module but rather the Venly login.
Im on silicon macbook using chrome

davidzwfu commented:
> I get an error like this with my pin of 6 digits <img alt="Screenshot 2023-05-16 at 10 09 33 AM" width="375" src="https://user-images.githubusercontent.com/24457880/238706488-f27a449c-bf48-4134-ba0d-fcec3011c9f7.png"> Also, when my pin was set to 4 digits I got a similar error that said it needed to be 6 digits Couldn't connect to the account I made through the main Venly page at all that had a wallet setup (maybe production) Doesnt seem to be an issue with the web3-onboard wallet module but rather the Venly login. Im on silicon macbook using chrome

It sounds like you might have mixed up the environments. You can tell the env you're on by looking at the url.

For example:
wallet.venly.io vs wallet-staging.venly.io
login.venly.io vs login-staging.venly.io

Can you verify which one the error is happening on?

Adamj1232 commented:
> 

This is happening on staging from the web3-onboard internal svelte demo app.
This occurs after I have setup an account through the modal that is popped from the demo. I havent altered anything within the internal demo from your fork/branch which is where I am testing from.

davidzwfu commented:
@Adamj1232 I've updated the docs with information about networks/environments and modified the Venly provider to throw an error when attempting to switch between environments (support for switching between environments is in the works). 

Regarding your issues, last week we had some data migration for our staging environment which was causing weird errors for newly created wallets. If you're still having issues please let me know the wallet addresses you are using so I can look into it.

Also your clientId is now enabled for use on production at the domains onboard.blocknative.com and reactdemo.blocknative.com

Adamj1232 commented:
@davidzwfu sounds good and thank you! I will give it a run through some testing now and update here as that progresses.

Adamj1232 commented:
@davidzwfu Can you give me write access on this fork?
When I pulled down this branch and ran the internal demo I noted 
- The Venly package isn't added to the `packages/demo` where the testing should take place by running `yarn dev` from the root of the project
- Sepolia test network wasnt supported
- When switching to a to mainnet or an L2 while on the venly staging env there is no feedback so it doesnt look like chain handling is correct
- When switching to Polygon - Mumbai and signing a typed message I get a console error which is fine if its not supported but then the chain switches back to Goerli
- Signing a message on Goerli gives me this error still:
<img width="362" alt="Screenshot 2023-05-22 at 10 04 47 AM" src="https://github.com/blocknative/web3-onboard/assets/24457880/8b66174b-8f99-48e2-bc09-9d889ea90d5c">

Adamj1232 commented:
@davidzwfu thank you for the access - I pushed some changes to the internal demo for testing. As stated above I am still seeing some errors on the wallet side - if they are expected thats fine just wanted to raise with you/your team in case they were an issue

Adamj1232 commented:
> > @davidzwfu thank you for the access - I pushed some changes to the internal demo for testing. As stated above I am still seeing some errors on the wallet side - if they are expected thats fine just wanted to raise with you/your team in case they were an issue
> 
> Could you please copy for me the full address of the wallet you were having issues with? I'll have the back-end team look into it.

@davidzwfu the staging address Im using for testing is 0xd36c45A1B04e8a6140f411D0FEF662F65Cc126F0

davidzwfu commented:
@Adamj1232 Updated to our latest release which supports switching between main and test networks (provided that your domain is configured for production use). Also updated the docs with our list of supported networks.

Regarding your previous issues:
- I've gone ahead and reset your staging wallet so you should be able to properly sign messages now (turns out your wallet's metadata was corrupted in the previous migration)
- At the moment we only support Goerli but have plans to migrate to Sepolia following deprecation
- I can't seem to get any error feedback when switching to an unsupported chain, despite returning the appropriate RPC error on the provider side.
- Still unknown to me why signing typed messages in Polygon reverts the chain back to Goerli. Logic for switching chains is technically only called when 'wallet_switchEthereumChain' method is requested.

Adamj1232 reviewed (approved):
Looks good! Thank you for the collaborative effort getting Venly support through! If you would be interested in some co marketing please let me know and I can put you in touch with the BD leads here.

Adamj1232 commented:
@davidzwfu we are seeing an error when adding Venly to our docs/demo site  - please pull down the latest develop, `cd docs/` `yarn`, comment in venly within the wallets array [here](https://github.com/blocknative/web3-onboard/blob/a2ffb41a37e6131fe7f76357d2755cf9acabc66c/docs/src/lib/services/onboard.js#L141) and `yarn dev` and select venly to see error `require is not defined`
Unfortunately until we have resolution here we cannot promote the Venly package or support

davidzwfu commented:
> @davidzwfu we are seeing an error when adding Venly to our docs/demo site - please pull down the latest develop, `cd docs/` `yarn`, comment in venly within the wallets array [here](https://github.com/blocknative/web3-onboard/blob/a2ffb41a37e6131fe7f76357d2755cf9acabc66c/docs/src/lib/services/onboard.js#L141) and `yarn dev` and select venly to see error `require is not defined` Unfortunately until we have resolution here we cannot promote the Venly package or support

I managed to fix that error by removing `'http'` from line 31 of vite.config.js, but now I'm getting a different error about my imports being undefined.

Adamj1232 commented:
@davidzwfu interesting - I had initially added http to fix this error when opening Venly
We are on node 16.18.1 
<img width="908" alt="Screenshot 2023-06-02 at 10 12 09 AM" src="https://github.com/blocknative/web3-onboard/assets/24457880/b5bab3c8-eb8c-48a7-a01b-d65a070db36c">
